### PR TITLE
Added PerfectGroup Function

### PR DESCRIPTION
**Pull request by divyanshu132**
#### Brief description of what is fixed or changed
Function `is_perfect` has been added to check if a group is `perfect group` or not.

#### Release Notes

*  combinatorics
   *  Added function for perfect group in perm_groups.py

### CODE REVIEW

**Comment by divyanshu132** [comment]
@jksuom please review.

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -1643,6 +1644,25 @@ def contains(self, g, strict=True):
...
+        >>> from sympy.combinatorics import Permutation
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> a = Permutation(1,2,3)(4,5)
+        >>> b = Permutation(1,2,3,4,5)
+        >>> G = PermutationGroup([a, b])
+        >>> G.is_perfect
+        False
+
+        """
+        return self == self.derived_subgroup()
```
Comment (jksuom) [inline sympy/combinatorics/perm_groups.py]: This could be used to set `_is_perfect`. Then there would be no need to recompute the derived subgroup.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: I didn't get you doing this `_is_perfect = self.derived_subgroup()` will make it better?

Comment (jksuom) [reply sympy/combinatorics/perm_groups.py]: If you set `_is_perfect` when 'is_perfect` is called for the first time, it is not necessary to recompute the derived subgroup.

    if self._is_perfect is None:
        set self._is_perfect
    return self._is_perfect

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: So, no need to compute derived_subgroup at all.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: @jksuom it shows error:
```
    set (self._is_perfect)
TypeError: 'NoneType' object is not iterable
```

Comment (jksuom) [reply sympy/combinatorics/perm_groups.py]: That was not valid Python code but pseudocode for the actual assignment of the proper value to `self._is_perfect`. For instance, if the proper value is `self == self.derived_subgroup()`, then that should be assigned.
`self._is_perfect` will the function as a cache. Computing `self.derived_subgroup()` is time consuming and that should not be repeated several times for asking if the group is perfect.

Comment (divyanshu132) [reply sympy/combinatorics/perm_groups.py]: Okay, that's why I got confused that how come we don't need to use `derived_subgroup()` at all and directly make is_perfect True.

**Comment by divyanshu132** [comment]
@jksuom can this be merged.

File: sympy/combinatorics/tests/test_perm_groups.py [hunk 2]
```diff
@@ -919,3 +919,13 @@ def test_elementary():
...
     assert H.is_elementary(2) == True
+
+
+def test_perfect():
+    G = AlternatingGroup(3)
+    assert G.is_perfect == False
+    G = AlternatingGroup(5)
+    assert G.is_perfect == True
+    a = Permutation(1)
+    G = PermutationGroup([a])
```
Comment (jksuom) [inline sympy/combinatorics/tests/test_perm_groups.py]: Perhaps this trivial group could be omitted.

Comment (divyanshu132) [reply sympy/combinatorics/tests/test_perm_groups.py]: Done!

**Comment by divyanshu132** [comment]
@jksuom please have a look!

**Comment by jksuom** [comment]
Thanks, this looks good.

**Comment by divyanshu132** [comment]
Thanks.

**Pull request merged**